iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/StatementAnalyzer.java

```diff
@@ -362,8 +366,50 @@ protected Scope visitShowIndex(ShowIndex node, Optional<Scope> context) {
     }
 
     @Override
-    protected Scope visitUpdate(Update node, Optional<Scope> context) {
-      throw new SemanticException("Update statement is not supported yet.");
+    protected Scope visitUpdate(final Update node, final Optional<Scope> context) {
+      queryContext.setQueryType(QueryType.WRITE);
+      final TranslationMap translationMap = analyzeTraverseDevice(node, context, true);
+      final TsTable table =
+          DataNodeTableCache.getInstance().getTable(node.getDatabase(), node.getTableName());
```

Comment:
Avoid directly call the singleton.getInstance, using `metadata.getTableSchema` instead. And actually you need to take care of what if the returned Optional.empty(), which means there is no such table.